tests/modules/accounting/test_withdrawal_unit.py

```diff
@@ -22,24 +24,63 @@ def past_blockstamp(web3, consensus_client):
 
 
 @pytest.fixture()
-def subject(web3, past_blockstamp, chain_config, frame_config, contracts, keys_api_client, consensus_client):
+def subject(
+    web3,
+    past_blockstamp,
+    chain_config,
+    frame_config,
+    contracts,
+    keys_api_client,
+    consensus_client
+):
     return Withdrawal(web3, past_blockstamp, chain_config, frame_config)
 
 
-def test_returns_zero_if_no_unfinalized_requests(subject):
-    subject._has_unfinalized_requests = Mock(return_value=False)
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    ("is_paused", "has_unfinalized_requests", "expected_result"),
+    [
+        (False, True, 100),
+        (False, False, 0),
+        (True, True, 0)
+    ]
+)
+def test_returns_zero_if_no_unfinalized_requests(
+    subject: Withdrawal,
+    is_paused: bool,
+    has_unfinalized_requests: bool,
+    expected_result: int,
+):
+    subject._fetch_is_paused = Mock(return_value=is_paused)
+    subject._has_unfinalized_requests = Mock(return_value=has_unfinalized_requests)
     subject._get_available_eth = Mock(return_value=0)
+    subject.safe_border_service.get_safe_border_epoch = Mock(return_value=0)
+    subject._fetch_last_finalizable_request_id = Mock(return_value=expected_result)
 
     result = subject.get_next_last_finalizable_id(True, 100, 0, 0)
 
-    assert result == 0
+    assert result == expected_result
 
 
-def test_returns_last_finalizable_id(subject):
-    subject._has_unfinalized_requests = Mock(return_value=True)
-    subject._get_available_eth = Mock(return_value=100)
+@pytest.mark.unit
+@pytest.mark.parametrize(("last_finalized_id", "last_requested_id"), [(2, 1), (1, 1)])
+def test_has_unfinalized_requests(subject: Withdrawal, last_finalized_id: int, last_requested_id: int):
+    subject._fetch_last_finalized_request_id = Mock(return_value=last_finalized_id)
+    subject._fetch_last_request_id = Mock(return_value=last_requested_id)
 
-    subject.safe_border_service.get_safe_border_epoch = Mock(return_value=0)
-    subject._fetch_last_finalizable_request_id = Mock(return_value=1)
+    assert subject._has_unfinalized_requests() == (last_finalized_id > last_requested_id)
+
+
+@pytest.mark.unit
+@given(strategies.lists(strategies.integers(min_value=0), min_size=4, max_size=4))
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+def test_get_available_eth(
+    subject: Withdrawal,
+    params: list[int]
```

Comment:
hypothesis looks awesome!

could you unpack params to the local variables? params by index looks not enough obvious